src/EFCore.PG/Query/ExpressionVisitors/NpgsqlSqlTranslatingExpressionVisitor.cs

```diff
@@ -86,33 +63,15 @@ public class NpgsqlSqlTranslatingExpressionVisitor : SqlTranslatingExpressionVis
             : base(dependencies, queryModelVisitor, targetSelectExpression, topLevelPredicate, inProjection)
             => _queryModelVisitor = queryModelVisitor;
 
-        /// <inheritdoc />
-        protected override Expression VisitSubQuery(SubQueryExpression expression)
-            => base.VisitSubQuery(expression) ?? VisitLikeAnyAll(expression) ?? VisitEqualsAny(expression);
+        #region Overrides
 
         /// <inheritdoc />
         protected override Expression VisitBinary(BinaryExpression expression)
-        {
-            if (expression.NodeType == ExpressionType.ArrayIndex)
-            {
-                var properties = MemberAccessBindingExpressionVisitor.GetPropertyPath(
-                    expression.Left, _queryModelVisitor.QueryCompilationContext, out _);
-                if (properties.Count == 0)
-                    return base.VisitBinary(expression);
-                var lastPropertyType = properties[properties.Count - 1].ClrType;
-                if (lastPropertyType.IsArray && lastPropertyType.GetArrayRank() == 1)
-                {
-                    var left = Visit(expression.Left);
-                    var right = Visit(expression.Right);
-
-                    return left != null && right != null
-                        ? Expression.MakeBinary(ExpressionType.ArrayIndex, left, right)
-                        : null;
-                }
-            }
-
-            return base.VisitBinary(expression);
-        }
+            => expression.NodeType is ExpressionType.ArrayIndex && IsSafeToVisit(expression, Context)
```

Comment:
I identify with the desire to use expression-bodied methods, but the `IsSafeToVisit()` method is called only from this method and seems pretty specific to it... So I'd rather this method included its "safety checks" within itself rather than an additional method.

In addition there's multidimensional arrays... Are we supposed to be using `ArrayIndex` for single-dimensional and `ArrayAccess` for multi-dimensional? I understand it may not matter because the expression tree doesn't actually get executed in .NET - it gets translated to SQL - but I think it's still important to keep accurate on these things.

It's also OK to not support multidimensional arrays all in this PR (the current code doesn't).